src/osd/scrubber/ScrubStore.cc

```diff
@@ -15,60 +15,62 @@ using std::vector;
 using ceph::bufferlist;
 
 namespace {
+
+/// the 'to_str()' representation of an hobject, if created with the specific
+/// parameters, prefixed with 'prefix'.
+std::string virtual_hobject_w_prefix(std::string_view prefix,
+                                     const std::string& key, snapid_t snap,
+                                     uint32_t hash, int64_t pool)
+{
+  return fmt::format("{}_{}", prefix,
+                     hobject_t(object_t(), key, snap, hash, pool, "").to_str());
+}
+
+/// the 'to_str()' representation of an hobject, if created with the specific
+/// parameters, prefixed with 'prefix'.
+std::string virtual_hobject_w_prefix(std::string_view prefix,
+                                     const librados::object_id_t& oid,
+                                     uint32_t hash, int64_t pool) {
+  return fmt::format("{}_{}", prefix,
+                     hobject_t(object_t{oid.name}, oid.locator, oid.snap, hash,
+                               pool, oid.nspace)
+                         .to_str());
+}
+
 string first_object_key(int64_t pool)
 {
-  auto hoid = hobject_t(object_t(), "", CEPH_NOSNAP, 0x00000000, pool, "");
-  hoid.build_hash_cache();
```

Comment:
OK, I think the relevant ctors of `hobject_t` call `build_hash_cache()` on their own.

```cpp
  hobject_t(const object_t& oid, const std::string& key, snapid_t snap,
            uint32_t hash, int64_t pool, const std::string& nspace)
    : oid(oid), snap(snap), hash(hash), max(false),
      pool(pool), nspace(nspace),
      key(oid.name == key ? std::string() : key) {
    build_hash_cache();
  }
```

Reply:
Yes - they do.